pymatgen/io/vasp/inputs.py

```diff
@@ -443,7 +443,7 @@ def from_str(cls, data, default_names=None, read_velocities=True) -> Poscar:
                 selective_dynamics.append([value == "T" for value in tokens[3:6]])
 
         # Warn when ALL degrees of freedom relaxed (#3539)
-        if has_selective_dynamics and all(all(i is True for i in in_list) for in_list in selective_dynamics):
+        if selective_dynamics is not None and all(all(i is True for i in in_list) for in_list in selective_dynamics):
```

Comment:
Use `selective_dynamics` instead of `has_selective_dynamics` to avoid: `pymatgen\io\vasp\inputs.py:443: error: Item "None" of "list[Any] | None" has no attribute "append"  [union-attr]`